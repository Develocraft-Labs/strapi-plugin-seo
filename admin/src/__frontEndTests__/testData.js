import trads from "../translations";

export const id = 1,
  firstname = "John",
  lastname = "Doe",
  email = "John.Doe@companyA.com",
  title = "title",
  created_by = {
    id,
    email,
    firstname,
    lastname,
  },
  created_at = "2021-10-12T11:49:01.042Z",
  updated_at = "2021-10-15T09:43:36.344Z",
  updated_by = { email, firstname, id, isActive: true, lastname },
  seo = {
    canonical: "https://www.companyA.com/articles",
    created_at: "2021-10-12T11:55:00.595Z",
    created_by: 1,
    id: 1,
    jsonLd: {},
    metaDescription: "Artists Desc",
    metaTitle: "Artists Title",
    ogImage: "Artist og image",
    ogTitle: "Artists Og Title",
    ogType: "Artists Og Type",
    ogUrl: "Artists og url",
    title: "Artists Seo",
    updated_at,
    updated_by: 1,
  },
  resourceData = {
    title,
    id,
    created_by,
  },
  collectionName = "Artists",
  pagination = { page: 1, total: 13, pageSize: 5, pageCount: 3 },
  uid = "application::artists.artists",
  fullResults = [
    { created_at, created_by, id, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 2, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 3, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 4, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 5, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 6, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 7, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 8, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 9, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 10, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 11, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 12, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 13, seo, title, updated_at, updated_by },
  ],
  seoId = 1,
  apiID = "artists",
  attributes = {
    seo: {
      dominant: false,
      nature: "oneWay",
      plugin: "seo",
      private: false,
      target: "plugins::seo.seo",
      unique: false,
    },
    title,
  },
  schema = {
    attributes,
    collectionName,
    connection: "default",
    description: "",
    draftAndPublish: true,
    kind: "collectionType",
    name: "Artists",
    pluginOptions: {},
    restrictRelationsTo: null,
    visible: true,
  },
  resourceSchema = { collectionName, schema },
  pluginUID = "plugins::seo.seo",
  projectCollectionTypes = [{ apiID, schema, uid }],
  selectedSeo = {
    canonical: "https://www.companyA.com/articles",
    created_at: "2021-10-12T11:55:00.595Z",
    created_by: created_by,
    id: 1,
    jsonLd: {},
    metaDescription: "Artists Desc",
    metaTitle: "Artists Title",
    ogImage: "Artist og image",
    ogTitle: "Artists Og Title",
    ogType: "Artists Og Type",
    ogUrl: "Artists og url",
    title: "Artists Seo",
    updated_at,
    updated_by: updated_by,
  },
  refreshOptions = () => {},
  metaTitle = "meta title",
  metaDescription = "meta description",
  canonical = "canonical",
  handleDeleteSeo = () => {},
  handleEditSeo = () => {},
  defaultLocale = "en",
  userEnabledLocales = [
    {
      code: "en",
      created_at: "2021-12-20T09:03:53.724Z",
      id: 1,
      isDefault: true,
      name: "English (en)",
      updated_at: "2021-12-20T09:03:53.724Z",
    },
  ],
  seos = [seo, fullResults];

export const resourceComponentTestData = {
  resourceData,
  resourceSchema,
};

export const seoFormComponentTestData = {
  pluginUID,
  seoId,
  projectCollectionTypes,
  resource: resourceComponentTestData,
};

export const contentTypesListComponentTestData = [
  {
    collectionName,
    fullResults,
    pagination,
    uid,
  },
];

export const translationPickerComponentTestData = {
  translations: trads,
  handleTranslation: () => {},
  value: "",
};

export const contentTypesItemComponentTestData = {
  uid,
  fullResults,
  pagination,
  collectionName,
};
export const previewComponentTestData = {
  selectedSeo,
  pluginUID,
  refreshOptions,
  projectCollectionTypes,
  handleDeleteSeo,
};

export const tableComponentTestData = {
  seos: { collectionName, fullResults, uid, pagination },
  handleDeleteSeo,
  handleEditSeo,
  userEnabledLocales,
  uid,
  defaultLocale,
};
