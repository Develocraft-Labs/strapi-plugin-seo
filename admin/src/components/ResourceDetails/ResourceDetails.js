import React from "react";
import { capitalize } from "lodash";
import { Column, ellipsisCss, EvenColumn } from "../ui/common";
import styled from "styled-components";

const CollectionName = styled.h3`
  width: 100%;
  margin: 0px;
  font-size: 1.3rem;
  font-weight: 400;
  line-height: normal;
  color: rgb(120, 126, 143);
  ${ellipsisCss}
`;

/**
 * Component that renders a Content Types resource details.
 * @returns {JSX.Element} Resource details.
 */
const ResourceDetails = ({ resource: { resourceData, resourceSchema } }) => {
  const { collectionName } = resourceSchema.schema;
  const { title } = resourceData;

  return (
    <Column>
      <EvenColumn>
        {!!title && <h2 data-testid="resource-title">{capitalize(title)}</h2>}
        <CollectionName data-testid="resource-collection-title">{`API ID : ${collectionName}`}</CollectionName>
      </EvenColumn>
    </Column>
  );
};

export default ResourceDetails;
