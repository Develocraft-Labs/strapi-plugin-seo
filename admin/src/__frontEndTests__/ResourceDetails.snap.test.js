/**
 * @jest-environment jsdom
 */

import React from "react";
import renderer from "react-test-renderer";

import ResourceDetails from "../components/ResourceDetails/ResourceDetails";
import { resourceComponentTestData } from "./testData";

describe("SNAP TESTING RESOURCE-DETAILS COMPONENT", () => {
  it("Should Render Successfully", () => {
    const component = renderer.create(
      <ResourceDetails resource={resourceComponentTestData} />
    );
    const tree = component.toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
