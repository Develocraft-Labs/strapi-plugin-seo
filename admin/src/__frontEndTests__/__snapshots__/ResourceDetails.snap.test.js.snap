// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAP TESTING RESOURCE-DETAILS COMPONENT Should Render Successfully 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c2 {
  width: 100%;
  margin: 0px;
  font-size: 1.3rem;
  font-weight: 400;
  line-height: normal;
  color: rgb(120,126,143);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c0 c1"
  >
    <h2
      data-testid="resource-title"
    >
      Title
    </h2>
    <h3
      className="c2"
      data-testid="resource-collection-title"
    >
      API ID : Artists
    </h3>
  </div>
</div>
`;
